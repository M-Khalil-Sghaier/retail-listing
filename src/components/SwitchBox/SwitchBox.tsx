import React from "react";
import { Switch } from "@headlessui/react";

type SwitchBoxProps = {
  label: string;
  enabled: boolean;
  onChange: (checked: boolean) => void;
};

const SwitchBox = ({ label, enabled, onChange }: SwitchBoxProps) => {
  return (
    <Switch.Group>
      <div className="flex items-center">
        <Switch.Label className="mr-4">{label}</Switch.Label>
        <Switch
          checked={enabled}
          onChange={onChange}
          className={`${
            enabled ? "bg-red-600" : "bg-gray-200"
          } relative inline-flex items-center h-6 rounded-full w-11 transition-colors focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-red-100`}
          data-testid="switch-button"
        >
          <span
            data-testid={`switch-${enabled ? "on" : "off"}`}
            className={`${
              enabled ? "translate-x-6" : "translate-x-1"
            } inline-block w-4 h-4 transform bg-white rounded-full transition-transform`}
          />
        </Switch>
      </div>
    </Switch.Group>
  );
};

export default SwitchBox;
