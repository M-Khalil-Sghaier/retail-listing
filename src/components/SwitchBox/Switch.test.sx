import { render, screen, waitFor } from "@testing-library/react";
import SwitchBox from "./SwitchBox";

describe("SwitchBox", () => {
  describe("Render", () => {
    it("Should render off by default", () => {
      const { container } = render(<SwitchBox label="Test" />);
      expect(screen.queryByTestId("switch-off")).toBeInTheDocument;
      expect(container).toMatchSnapshot();
    });
    it("Should render on by default", () => {
      render(<SwitchBox label="Test" enabled={true} />);
      expect(screen.queryByTestId("switch-on")).toBeInTheDocument;
    });
  });
  describe("Behavior", () => {
    it("Should toggle its state", () => {
      const onChange = jest.fn();
      render(<SwitchBox label="Test" onChange={onChange} />);
      expect(screen.getByTestId("switch-off")).toBeInTheDocument;
      const switchButton = screen.getByTestId("switch-button");
      switchButton.click();
      expect(screen.queryByTestId("switch-on")).toBeInTheDocument;
      expect(onChange).toHaveBeenCalledTimes(1);
    });
  });
});
