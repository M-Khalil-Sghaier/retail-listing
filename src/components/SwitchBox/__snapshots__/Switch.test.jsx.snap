// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwitchBox Render Should render off by default 1`] = `
<div>
  <div
    class="flex items-center"
  >
    <label
      class="mr-4"
      id="headlessui-label-1"
    >
      Test
    </label>
    <button
      aria-labelledby="headlessui-label-1"
      class="bg-gray-200 relative inline-flex items-center h-6 rounded-full w-11 transition-colors focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-red-100"
      data-testid="switch-button"
      id="headlessui-switch-2"
      role="switch"
      tabindex="0"
      type="button"
    >
      <span
        class="translate-x-1 inline-block w-4 h-4 transform bg-white rounded-full transition-transform"
        data-testid="switch-off"
      />
    </button>
  </div>
</div>
`;
